src/main/java/edu/kit/ipd/crowdcontrol/objectservice/rest/resources/ExperimentResource.java

```diff
@@ -126,45 +128,80 @@ public Experiment put(Request request, Response response) {
 
         record.setDescription(Template.apply(rawDescription, placeholders));
 
-        int id = experimentOperations.insertNewExperiment(record);
+        int id;
 
-        List<TagRecord> tags = TagConstraintTransformer.getTags(experiment, id);
-        List<ConstraintRecord> constraints = TagConstraintTransformer.getConstraints(experiment, id);
+        try {
+            id = experimentOperations.insertNewExperiment(record);
+        } catch (DataAccessException e) {
+            throw new InternalServerErrorException("Experiment could not be stored.", e);
+        }
 
-        tags.stream()
-                .filter(tagRecord -> !tagRecord.getTag().isEmpty())
-                .forEach(tagConstraintsOperations::insertTag);
+        try {
+            try {
+                List<TagRecord> tags = TagConstraintTransformer.getTags(experiment, id);
```

Comment:
I am quite not sure if this should be its own function which creates the info of a expeirment, makes reading easier